coding_bestpractices.rst

```diff
@@ -30,544 +30,391 @@ Bad:
-                   break
-           endswitch
+       switch(ba.eventCode)
+           case 2:
+               DoStuff(ba.win)
+               break
+       endswitch
 
-           return 0
-       End
+       return 0
+   End
 
 Reasons:
 
 -  Less error prone (the top window could be different if ButtonProc is
    called programmatically)
-
 -  Easily expandable to other events
-
 -  Faster as only a reference to the structure must be passed to the
    function
 
-Avoid magic numbers
--------------------
-
-Bad:
-
-.. code:: igor
-
-       settings[11][] = height
-
-Good:
-
-.. code:: igor
-
-       // either with file level constants
-       static Constant HEIGHT_ROW = 11
-
-       Function DoStuff()
-
-           settings[HEIGHT_ROW][] = height
-
-       End
-
-       // or dimension labels, use SetDimLabel on the wave before
-       settings[%height][] = height
-
-       // or a set function (rarely an appropriate choice)
-       Function SetHeight(settings, height)
-           WAVE settings
-           variable height
-
-           settings[11][] = height
-       End
-
-       SetHeight(settings, height)
-
-Reasons:
-
--  Better readability
-
--  Less error prone
-
 Unused parameters
------------------
+^^^^^^^^^^^^^^^^^
 
 Bad:
 
-.. code:: igor
+.. code-block:: igor
 
-       Function/S GetPath(param)
-           string param
+   Function/S GetPath(param)
+       string param
 
-           return "root:myFolder"
-       End
+       return "root:myFolder"
+   End
 
 Good:
 
-.. code:: igor
+.. code-block:: igor
 
-       Function/S GetPath()
-           return "root:myFolder"
-       End
+   Function/S GetPath()
+       return "root:myFolder"
+   End
 
 Reasons:
 
--  Better readability
+- Better readability
+- Does not fake a dependency of the function upon the parameter
 
--  Does not fake a dependency of the function upon the parameter
+.. note::
 
-Note: Unused function variables and file level constants should also be
-avoided. As there is currently no support from Igor Pro in doing that,
-visual inspection must be performed.
+   Unused function variables and file level constants should also be
+   avoided. As there is currently no support from Igor Pro in doing that,
+   visual inspection must be performed.
 
 Avoid function calls in loop statements
----------------------------------------
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Bad:
 
-.. code:: igor
+.. code-block:: igor
 
-       for(i = 0; i < ItemsInList(list); i += 1)
-           // code
-       endfor
+   for(i = 0; i < ItemsInList(list); i += 1)
+       // code
+   endfor
 
 Good:
 
-.. code:: igor
+.. code-block:: igor
 
-       numItems = ItemsInList(list)
-       for(i = 0; i < numItems ; i += 1)
-           // code
-       endfor
+   numItems = ItemsInList(list)
+   for(i = 0; i < numItems ; i += 1)
+       // code
+   endfor
 
 Reasons:
 
--  Faster, the bad example calls ``ItemsInList`` every time the loop
-   condition is executed
-
-Performance tests
```

Comment:
The performance tests are removed? WM recommends in its documentation the use of numeric indexes because dimension labels are way slower.